boards/tutorials/nrf52840dk-dynamic-apps-and-policies/src/main.rs

```diff
@@ -0,0 +1,649 @@
+            nrf52840::adc::AdcChannelSetup::new(nrf52840::adc::AdcChannel::AnalogInput5),
+            nrf52840::adc::AdcChannelSetup::new(nrf52840::adc::AdcChannel::AnalogInput6),
+            nrf52840::adc::AdcChannelSetup::new(nrf52840::adc::AdcChannel::AnalogInput7),
+        ]
+    );
+    let adc = components::adc::AdcDedicatedComponent::new(
+        &base_peripherals.adc,
+        adc_channels,
+        board_kernel,
+        capsules_core::adc::DRIVER_NUM,
+    )
+    .finalize(components::adc_dedicated_component_static!(
+        nrf52840::adc::Adc
+    ));
+
+    //--------------------------------------------------------------------------
+    // VIRTUAL FLASH
+    //--------------------------------------------------------------------------
+
+    let mux_flash = components::flash::FlashMuxComponent::new(&nrf52840_peripherals.nrf52.nvmc)
+        .finalize(components::flash_mux_component_static!(
+            nrf52840::nvmc::Nvmc
+        ));
+
+    // Create a virtual flash user for dynamic binary storage
+    let virtual_flash_dbs = components::flash::FlashUserComponent::new(mux_flash).finalize(
+        components::flash_user_component_static!(nrf52840::nvmc::Nvmc),
+    );
+
+    // Create a virtual flash user for nonvolatile
+    let virtual_flash_nvm = components::flash::FlashUserComponent::new(mux_flash).finalize(
+        components::flash_user_component_static!(nrf52840::nvmc::Nvmc),
+    );
+
+    //--------------------------------------------------------------------------
+    // NONVOLATILE STORAGE
+    //--------------------------------------------------------------------------
+
+    // 32kB of userspace-accessible storage, page aligned:
+    kernel::storage_volume!(APP_STORAGE, 32);
+
+    let nonvolatile_storage = components::isolated_nonvolatile_storage::IsolatedNonvolatileStorageComponent::new(
+        board_kernel,
+        capsules_extra::isolated_nonvolatile_storage_driver::DRIVER_NUM,
+        virtual_flash_nvm,
+        core::ptr::addr_of!(APP_STORAGE) as usize,
+        APP_STORAGE.len()
+    )
+    .finalize(components::isolated_nonvolatile_storage_component_static!(
+        capsules_core::virtualizers::virtual_flash::FlashUser<'static, nrf52840::nvmc::Nvmc>,
+        { components::isolated_nonvolatile_storage::ISOLATED_NONVOLATILE_STORAGE_APP_REGION_SIZE_DEFAULT }
+    ));
+
+    //--------------------------------------------------------------------------
+    // NRF CLOCK SETUP
+    //--------------------------------------------------------------------------
+
+    nrf52_components::NrfClockComponent::new(&base_peripherals.clock).finalize(());
+
+    //--------------------------------------------------------------------------
+    // SCREEN
+    //--------------------------------------------------------------------------
+
+    const SCREEN_I2C_SDA_PIN: Pin = Pin::P1_10;
+    const SCREEN_I2C_SCL_PIN: Pin = Pin::P1_11;
```

Comment:
oof; not a problem this PR needs to fix, but this is the third time we're copy-pasting this block; we should componentize this somehow

(non-blocking comment)